src/components/OpportunityCard/OpportunityCard.tsx

```diff
@@ -114,11 +114,24 @@ const MoreActionsDropdown = ({ compareTo }: { compareTo?: To }) => {
   const { t } = useTranslation();
   const [open, isOpen] = React.useState(false);
   const listId = React.useId();
+  const actionButtonRef = React.useRef<HTMLDivElement>(null);
 
   const onClose = React.useCallback(() => isOpen(false), []);
 
+  const handleClickOutside = (event: MouseEvent) => {
+    if (actionButtonRef.current && !actionButtonRef.current.contains(event.target as Node)) {
+      isOpen(false);
+    }
+  };
+  React.useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
```

Comment:
We can make this as a reusable hook, for e.g:

```tsx
// src/hooks/useClickOutside/useClickOutside.tsx
import React from 'react';

export const useOutsideClick = (handleOutsideClick: (event: MouseEvent) => void) => {
  const ref = React.useRef<HTMLDivElement>(null);

  React.useEffect(() => {
    const clickHandler = (event: MouseEvent) => {
      if (ref.current && !ref.current.contains(event.target as Node)) {
        handleOutsideClick(event);
      }
    };
    document.addEventListener('mousedown', clickHandler, true);
    return () => {
      document.removeEventListener('mousedown', clickHandler, true);
    };
  }, [ref, handleOutsideClick]);

  return ref;
};

```

Usage:

```tsx
  const actionButtonRef = useOutsideClick(() => isOpen(false));
// and using ref as before
```

Reply:
Created reusable hook, and also fixed problem when the menu button was clicked.